### PR TITLE
Drop type specs in RecoHI, RecoJets and RecoMET

### PR DESCRIPTION
**Pull request by NiharSaha**
#### PR description:
Drop type specs in Validation/RecoHI, Validation/RecoJets and  Validation/RecoMET python configuration files.

Central DQM migration campaign for drop type spces following
https://cms-sw.github.io/cms_coding_rules.html#6--packaging-rules-1
and update the safer syntax for existing parameters like: (1) drop type specifications where the original parameter exists. (2) using "clone" instead of "deepcopy" (3) move all parameters inside the clone

#### PR validation:
Tested in CMSSW_12_3_X via runTheMatrix.py -l limited -i all --ibeos

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

File: Validation/RecoMET/python/METRelValForDQM_cff.py [hunk 1]
```diff
@@ -35,18 +35,18 @@
...
 
-valCorrPfMetType1=corrPfMetType1.clone(jetCorrLabel = cms.InputTag('newAK4PFCHSL1FastL2L3Corrector'),
-                                       jetCorrLabelRes = cms.InputTag('newAK4PFCHSL1FastL2L3ResidualCorrector')
+valCorrPfMetType1=corrPfMetType1.clone(jetCorrLabel = 'newAK4PFCHSL1FastL2L3Corrector',
+                                       jetCorrLabelRes = 'newAK4PFCHSL1FastL2L3ResidualCorrector'
                                       )
 
 PfMetT1=pfMetT1.clone(srcCorrections = cms.VInputTag(
-        cms.InputTag('valCorrPfMetType1', 'type1')
+        ('valCorrPfMetType1', 'type1')
```
Comment (jfernan2) [inline Validation/RecoMET/python/METRelValForDQM_cff.py:43]: 'valCorrPfMetType1:type1'

File: Validation/RecoMET/python/METRelValForDQM_cff.py [hunk 2]
```diff
@@ -35,18 +35,18 @@
...
         ))
 
 PfMetT0pcT1=pfMetT0pcT1.clone(
     srcCorrections = cms.VInputTag(
-        cms.InputTag('corrPfMetType0PfCand'),
-        cms.InputTag('valCorrPfMetType1', 'type1')
+        'corrPfMetType0PfCand',
+        ('valCorrPfMetType1', 'type1')
```
Comment (jfernan2) [inline Validation/RecoMET/python/METRelValForDQM_cff.py:49]: 'valCorrPfMetType1:type1'

Comment (NiharSaha) [reply Validation/RecoMET/python/METRelValForDQM_cff.py:49]: @jfernan2  thanks. Let me change!

**Pull request closed without merging**